### ISSUE TITLE
Test output not consistently shown

### ISSUE THREAD

**Issue by Nepoxx** [alexfernandez/loadtest#42]
The test results are not consistently shown after running a loadtest. I am able to reproduce on multiple computers and multiple platforms (Windows 7 and Ubuntu 14).

```bash
user@ubuntu:~/nodetest/node$ loadtest -c 8 -t 40 --rps 200 http://127.0.0.1:80/api/hello/world.json
[Mon Jan 12 2015 09:52:54 GMT-0500 (EST)] INFO Requests: 0, requests per second: 0, mean latency: 0 ms
[Mon Jan 12 2015 09:52:59 GMT-0500 (EST)] INFO Requests: 885, requests per second: 177, mean latency: 0 ms
[Mon Jan 12 2015 09:53:04 GMT-0500 (EST)] INFO Requests: 1885, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:53:09 GMT-0500 (EST)] INFO Requests: 2885, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:53:14 GMT-0500 (EST)] INFO Requests: 3885, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:53:19 GMT-0500 (EST)] INFO Requests: 4885, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:53:24 GMT-0500 (EST)] INFO Requests: 5885, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:53:29 GMT-0500 (EST)] INFO Requests: 6885, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:53:34 GMT-0500 (EST)] INFO Requests: 7885, requests per second: 200, mean latency: 0 ms
user@ubuntu:~/nodetest/node$ loadtest -c 8 -t 40 --rps 200 http://127.0.0.1:80/api/hello/world.json
[Mon Jan 12 2015 09:53:41 GMT-0500 (EST)] INFO Requests: 0, requests per second: 0, mean latency: 0 ms
[Mon Jan 12 2015 09:53:46 GMT-0500 (EST)] INFO Requests: 890, requests per second: 178, mean latency: 0 ms
[Mon Jan 12 2015 09:53:51 GMT-0500 (EST)] INFO Requests: 1890, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:53:56 GMT-0500 (EST)] INFO Requests: 2890, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:54:01 GMT-0500 (EST)] INFO Requests: 3890, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:54:06 GMT-0500 (EST)] INFO Requests: 4890, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:54:11 GMT-0500 (EST)] INFO Requests: 5890, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:54:16 GMT-0500 (EST)] INFO Requests: 6890, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:54:21 GMT-0500 (EST)] INFO Requests: 7890, requests per second: 200, mean latency: 0 ms
user@ubuntu:~/nodetest/node$ loadtest -c 8 -t 40 --rps 200 http://127.0.0.1:80/api/hello/world.json
[Mon Jan 12 2015 09:54:28 GMT-0500 (EST)] INFO Requests: 0, requests per second: 0, mean latency: 0 ms
[Mon Jan 12 2015 09:54:33 GMT-0500 (EST)] INFO Requests: 894, requests per second: 179, mean latency: 0 ms
[Mon Jan 12 2015 09:54:38 GMT-0500 (EST)] INFO Requests: 1891, requests per second: 199, mean latency: 0 ms
[Mon Jan 12 2015 09:54:43 GMT-0500 (EST)] INFO Requests: 2894, requests per second: 201, mean latency: 0 ms
[Mon Jan 12 2015 09:54:48 GMT-0500 (EST)] INFO Requests: 3893, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:54:53 GMT-0500 (EST)] INFO Requests: 4892, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:54:58 GMT-0500 (EST)] INFO Requests: 5893, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:55:03 GMT-0500 (EST)] INFO Requests: 6894, requests per second: 200, mean latency: 0 ms
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO Target URL:          http://127.0.0.1:80/api/hello/world.json
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO Max time (s):        40
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO Concurrency level:   8
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO Agent:               none
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO Requests per second: 200
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO Completed requests:  7892
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO Total errors:        0
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO Total time:          40.003793234999996 s
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO Requests per second: 197
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO Total time:          40.003793234999996 s
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO Percentage of the requests served within a certain time
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO   50%      1 ms
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO   90%      2 ms
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO   95%      2 ms
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO   99%      3 ms
[Mon Jan 12 2015 09:55:08 GMT-0500 (EST)] INFO  100%      17 ms (longest request)
```

I'm not able to reproduce this issue consistently however, as shown above. The last output is missing a line, "Requests: X, requests per second: Y, mean latency: Z ms ", so my guess is that the loadtest is killed before it has time to output the results. This seems to happen less frequently with tests that take longer to run, further evidence towards my above hypothesis.

**Comment by alexfernandez** [comment]
I had never seen this on Debian or Mac OS X, but I have just now been able to reproduce it. I'll look into it.

**Comment by alexfernandez** [comment]
The issue appeared only with `--rps` since, as you guessed, the load tester counted 40 seconds but the timer didn't yet, and thus did not show the output. It should be solved in 1.2.5.

I'm closing this issue, please reopen if necessary.
